src/app/services/constraint.service.ts

```diff
@@ -119,8 +120,11 @@ export class ConstraintService {
   * when user defines a cohort or a combination of cohorts.
   */
   private _variables: Concept[] = [];
+  private _categorizedVariables: Array<CategorizedVariable> = [];
```

Comment:
Please separate cohort selection and variable selection into separate services.

Reply:
cohort selection?